src/main/target/JHEMCUH743HD/target.h

```diff
@@ -0,0 +1,221 @@
+/*
+ * This file is part of INAV.
+ *
+ * INAV is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * INAV is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with INAV.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define TARGET_BOARD_IDENTIFIER "JHEH743HD"
+#define USBD_PRODUCT_STRING "JHEMCUH743HD"
+
+
+
+#define USE_TARGET_CONFIG
+
+#define LED0 PE5
+#define LED1 PE4
+
+#define BEEPER PE3
+#define BEEPER_INVERTED
+
+// *************** IMU generic ***********************
+#define USE_DUAL_GYRO
+#define USE_TARGET_IMU_HARDWARE_DESCRIPTORS
+
+// *************** SPI1 IMU0 MPU6000 ****************
+#define USE_SPI
+#define USE_SPI_DEVICE_1
+#define SPI1_SCK_PIN PA5
+#define SPI1_MISO_PIN PA6
+#define SPI1_MOSI_PIN PA7
+
+#define USE_IMU_MPU6000
+#define IMU_MPU6000_ALIGN CW180_DEG
+#define MPU6000_SPI_BUS BUS_SPI1
+#define MPU6000_CS_PIN PA4
+
+#define USE_IMU_BMI270
+#define IMU_BMI270_ALIGN CW180_DEG
+#define BMI270_SPI_BUS BUS_SPI1
+#define BMI270_CS_PIN PA4
+
+
+#define USE_IMU_ICM42605
+#define IMU_ICM42605_ALIGN      CW180_DEG
+#define ICM42605_SPI_BUS        BUS_SPI1
+#define ICM42605_CS_PIN         PA4
+
+
+
+// *************** SPI2 OSD ***********************
+#define USE_SPI_DEVICE_2
+#define SPI2_SCK_PIN PB13
+#define SPI2_MISO_PIN PB14
+#define SPI2_MOSI_PIN PB15
+
+#define USE_MAX7456
+#define MAX7456_SPI_BUS BUS_SPI2
+#define MAX7456_CS_PIN PB12
+
+// *************** SPI3 FLASH ***********************
+#define USE_SPI_DEVICE_3
+#define SPI3_SCK_PIN PC10
+#define SPI3_MISO_PIN PC11
+#define SPI3_MOSI_PIN PB2
+#define SPI3_NSS_PIN PA15
+#define SPI3_SCK_AF GPIO_AF6_SPI3
+#define SPI3_MISO_AF GPIO_AF6_SPI3
+#define SPI3_MOSI_AF GPIO_AF7_SPI3
+
+#define M25P16_SPI_BUS BUS_SPI3
+#define M25P16_CS_PIN SPI3_NSS_PIN
+
+#define W25N01G_SPI_BUS BUS_SPI3
+#define W25N01G_CS_PIN SPI3_NSS_PIN
+
+#define USE_BLACKBOX
+#define USE_FLASHFS
+#define USE_FLASH_M25P16
+#define USE_FLASH_W25N01G
+#define ENABLE_BLACKBOX_LOGGING_ON_SPIFLASH_BY_DEFAULT
+
+// *************** I2C /Baro/Mag *********************
+#define USE_I2C
+#define USE_I2C_DEVICE_1
+#define I2C1_SCL PB6
+#define I2C1_SDA PB7
+
+#define USE_I2C_DEVICE_2
+#define I2C2_SCL PB10
+#define I2C2_SDA PB11
+
+#define USE_BARO
+#define BARO_I2C_BUS BUS_I2C1
+#define USE_BARO_BMP280
+#define USE_BARO_MS5611
+#define USE_BARO_DPS310
+#define USE_BARO_SPL06
+
+#define USE_MAG
+#define MAG_I2C_BUS BUS_I2C1
+#define USE_MAG_HMC5883
```

Comment:
Same as the others, #define USE_MAG_ALL instead of listing all compass chips manually.